modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -89,28 +108,72 @@
 
   private ExecutionMediator executionMediator;
   private OperationExecutor operationExecutor;
+  private PolicyManager policyManager;
   protected ReturnDelegate returnDelegate;
 
   public OperationMessageProcessor(ExtensionModel extensionModel,
                                    OperationModel operationModel,
                                    ConfigurationProvider configurationProvider,
                                    String target,
                                    ResolverSet resolverSet,
-                                   ExtensionManagerAdapter extensionManager) {
+                                   ExtensionManagerAdapter extensionManager,
+                                   PolicyManager policyManager) {
     super(extensionModel, operationModel, configurationProvider, extensionManager);
     this.extensionModel = extensionModel;
     this.operationModel = operationModel;
     this.resolverSet = resolverSet;
     this.target = target;
     this.entityMetadataMediator = new EntityMetadataMediator(operationModel);
+    this.policyManager = policyManager;
   }
 
   @Override
   public Event process(Event event) throws MuleException {
-    return (Event) withContextClassLoader(getExtensionClassLoader(), () -> {
+    return withContextClassLoader(getExtensionClassLoader(), () -> {
       Optional<ConfigurationInstance> configuration = getConfiguration(event);
-      ExecutionContextAdapter operationContext = createExecutionContext(configuration, event);
-      return doProcess(event, operationContext);
+
+      ComponentIdentifier operationIdentifier =
```

Comment:
can this logic be generified and abstracted to some reusable component. It seems to me like this algorithm will be  (maybe with minor changes) applied many times (it actually reminds me a lot of ExtensionFlowProcessingPhase) . Also, to have this plain here makes it hard to read and understand the actual OperationMP logic.